### ISSUE TITLE
Generate real Markdown dynamically

### ISSUE THREAD

**Issue by redactedscribe** [johansatge/obsidian-automatic-table-of-contents#10]
It would be great if the codeblock containing the settings for the TOC could be defined elsewhere, e.g. via the plugin settings page. Then a TOC with a predefined style could be inserted via the Command Palette. The benefit would be that we could insert real Markdown code which would therefore be visible and useable in other Markdown viewers, not only Obsidian, and for example online.

These TOCs inserted via the Command Palette should also be tracked and dynamically updated when headings are added or removed, again with real Markdown, otherwise it would be static and soon outdated.

It would also be cool if a TOC defined via a codeblock could be converted into real Markdown in case you want to export the file or something.

I don't have much experience with your plugin, but it seems like this feature request calls into question what purpose codeblock TOCs would serve if real inserted Markdown would behave the same but with the additional benefit of being portable.

In short, dynamic TOCs are great, but I'd much prefer if it was real Markdown code (VS Code has [a plugin](https://github.com/yzhang-gh/vscode-markdown#table-of-contents) which does this FR nicely).

Thanks!

**Comment by rajesh-s** [comment]
@redactedscribe did you find a workaround? I'm looking for the same and did not find a replacement.

**Comment by g-berthiaume** [comment]
+1 for generating real markdown. 
If this plugging or the obsidian's projects gets abandoned, I would like for my table of content to still exist.

Thanks for your work @johansatge !
